src/python_testing/TC_CADMIN_1_3_4.py

```diff
@@ -0,0 +1,365 @@
+from chip import ChipDeviceCtrl
+from chip.exceptions import ChipStackError
+from chip.interaction_model import Status
+from chip.tlv import TLVReader
+from matter_testing_infrastructure.chip.testing.matter_testing import (MatterBaseTest, TestStep, async_test_body,
+                                                                       default_matter_test_main)
+from mdns_discovery import mdns_discovery
+from mobly import asserts
+
+opcreds = Clusters.OperationalCredentials
+nonce = random.randbytes(32)
+
+
+class TC_CADMIN_1_3_4(MatterBaseTest):
+    async def get_fabrics(self, th: ChipDeviceCtrl) -> int:
+        OC_cluster = Clusters.OperationalCredentials
+        if th == self.th2:
+            th2_fabric_info = await th.ReadAttribute(nodeid=self.dut_node_id, fabricFiltered=True, attributes=[(0, OC_cluster.Attributes.Fabrics)])
+            th2_fabric_data = list(th2_fabric_info.values())[0]
+            th2_fabric_data = list(th2_fabric_data.values())[0]
+            fabric_info = vars(list(th2_fabric_data.values())[1][0])
+
+        else:
+            fabric_info = await self.read_single_attribute_check_success(dev_ctrl=th, fabric_filtered=True, cluster=OC_cluster, attribute=OC_cluster.Attributes.Fabrics)
+        return fabric_info
+
+    async def get_rcac_decoded(self, th: str) -> int:
+        csrResponse = await self.send_single_cmd(dev_ctrl=th, node_id=self.dut_node_id, cmd=opcreds.Commands.CSRRequest(CSRNonce=nonce, isForUpdateNOC=False))
+        TH_certs_real = await th.IssueNOCChain(csrResponse, self.dut_node_id)
+        th_rcac_decoded = TLVReader(TH_certs_real.rcacBytes).get()["Any"]
+        return th_rcac_decoded
+
+    async def get_txt_record(self):
+        discovery = mdns_discovery.MdnsDiscovery(verbose_logging=True)
+        comm_service = await discovery.get_commissionable_service(
+            discovery_timeout_sec=240,
+            log_output=False,
+        )
+        return comm_service
+
+    async def write_nl_attr(self, th: ChipDeviceCtrl, attr_val: object):
+        result = await th.WriteAttribute(nodeid=self.dut_node_id, attributes=[(0, attr_val)])
+        asserts.assert_equal(result[0].Status, Status.Success, f"{th} node label write failed")
+
+    async def read_nl_attr(self, th: ChipDeviceCtrl, attr_val: object):
+        try:
+            await th.ReadAttribute(nodeid=self.dut_node_id, attributes=[(0, attr_val)])
+        except Exception as e:
+            asserts.assert_equal(e.err, "Received error message from read attribute attempt")
+            self.print_step(0, e)
+
+    async def read_currentfabricindex(self, th: ChipDeviceCtrl) -> int:
+        cluster = Clusters.OperationalCredentials
+        attribute = Clusters.OperationalCredentials.Attributes.CurrentFabricIndex
+        current_fabric_index = await self.read_single_attribute_check_success(dev_ctrl=th, endpoint=0, cluster=cluster, attribute=attribute)
+        return current_fabric_index
+
+    def pics_TC_CADMIN_1_3(self) -> list[str]:
```

Comment:
can these two tests be combined into a single parameterized function?

Reply:
Looking at these two tests, from step 1 to step 7, the steps are basically identical with the exception of the command sent (open commissioning window vs. open basic commissioning window. Can definitely re-use some of this between tests.